src/msg/simple/Accepter.cc

```diff
@@ -239,21 +255,46 @@ void *Accepter::entry()
   ldout(msgr->cct,10) << "accepter starting" << dendl;
   
   int errors = 0;
+  int ch;
 
-  struct pollfd pfd;
-  pfd.fd = listen_sd;
-  pfd.events = POLLIN | POLLERR | POLLNVAL | POLLHUP;
+  struct pollfd pfd[2];
+  memset(pfd, 0, sizeof(pfd));
+  create_selfpipe(&shutdown_rd_fd, &shutdown_wr_fd);
```

Comment:
@liewegas 
Not as simple as I though, since it is a scoping problem.
The pipe is now local to entry(), and to use it in the poll() it needs to be known there.
Poll is destructive on the file descriptors struct, so you need to reinit the struct every time over.
And thus it has to become a private member of the class.
Lets see how that works out.